signer/tests/devenv/devenv.rs

```diff
@@ -0,0 +1,487 @@
+///
+/// In the test, it first process a deposit end to end, then forks bitcoin and
+/// invalidate the deposit tx (spending the same vin), and then mine some
+/// bitcoin blocks to resume the stacks mining. Finally, it checks that the mint
+/// stacks tx is marked as failed.
+///
+/// This test is pretty flaky: frequently the final checks fails as the stacks
+/// miner gets stuck (either keep building on the wrong consensus hash, with the
+/// signers rejecting the block, or complains about a missing PoX anchor).
+/// Sometime, the test fails but after some time (re-enabling the bitcoin miner,
+/// or just generating some blocks) the stacks miner resume producing blocks
+/// (with the deposit minting tx fail as expected).
+/// Raising `POX_PREPARE_LENGTH` and `POX_REWARD_LENGTH` (in devenv config)
+/// seems to help avoiding the PoX issue.
+#[test_log::test(tokio::test)]
+async fn orphaned_deposit() {
+    let stacks = StacksClient::new(Url::parse("http://127.0.0.1:20443").unwrap()).unwrap();
+    let (rpc, faucet) = regtest::initialize_blockchain_devenv();
+    let emily_client = EmilyClient::try_new(
+        &Url::parse("http://testApiKey@127.0.0.1:3031").unwrap(),
+        Duration::from_secs(1),
+        None,
+    )
+    .unwrap();
+
+    // Ensure there's a signers UTXO
+    let deployer = StacksAddress::from_string(DEVENV_DEPLOYER).unwrap();
+    let aggregate_key = stacks
+        .get_current_signers_aggregate_key(&deployer)
+        .await
+        .unwrap()
+        .expect("no signers aggregate key");
+    let signers_address = Address::from_script(
+        &aggregate_key.signers_script_pubkey(),
+        bitcoin::Network::Regtest,
+    )
+    .unwrap();
+    faucet.send_to(1000, &signers_address);
+    faucet.generate_block();
+
+    // Create a deposit request
+    let deposit_amount = 100_000;
+    let tx_fee = regtest::BITCOIN_CORE_FALLBACK_FEE.to_sat();
+
+    let depositor = Recipient::new(AddressType::P2tr);
+    let recipient: StacksPrincipal = fake::Faker.fake_with_rng(&mut OsRng);
+    dbg!(&recipient);
+
+    // Send some extra to ensure `make_deposit_request` doesn't create dust change
+    let depositor_fund_outpoint = faucet.send_to(deposit_amount * 2 + tx_fee, &depositor.address);
+    faucet.generate_block();
+
+    let depositor_utxo = Utxo {
+        txid: depositor_fund_outpoint.txid,
+        vout: depositor_fund_outpoint.vout,
+        script_pub_key: depositor.address.script_pubkey(),
+        descriptor: "".to_string(),
+        amount: Amount::from_sat(deposit_amount * 2 + tx_fee),
+        height: 0,
+    };
+    let max_fee = deposit_amount / 2;
+    let (deposit_tx, deposit_request, _) = make_deposit_request(
+        &depositor,
+        deposit_amount,
+        depositor_utxo.clone(),
+        max_fee,
+        aggregate_key.into(),
+        recipient.clone().into(),
+    );
+    rpc.send_raw_transaction(&deposit_tx).unwrap();
+    let block_deposit_tx = faucet.generate_block();
```

Comment:
Shouldn't we generate the block after submitting to Emily? I would think that it could fail if the signers were sufficiently fast (even though that's unlikely).

Reply:
Good point, I don't think it would fail too badly, but just to be sure I moved it after the creation on Emily.